### PR TITLE
Ansible Script to rollback migrations

### PR DESCRIPTION
**Pull request by maxrothman**
Configuration Pull Request
---

Make sure that the following steps are done before merging

  - [ ] @devops team member has commented with :+1:
  - [ ] are you adding any new default values that need to be overriden when this goes live?  
    - [ ] Open a ticket (DEVOPS) to make sure that they have been added to secure vars.
    - [ ] Add an entry to the CHANGELOG.

### CODE REVIEW

**Comment by macdiesel** [comment]
This is ready for review.

**Comment by macdiesel** [comment]
It is intended that this script will be called multiple times, once for each migration output from the migration stage.

File: playbooks/continuous_delivery/rollback_migrations.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+#                                         when APPLICATION_NAME is edxapp
+#   - EDX_PLATFORM_SETTINGS           - The settings to use for the edx platform {aws|devstack} DEFAULT: aws
+#   - HIPCHAT_TOKEN                   - API token to send messages to hipchat
+#   - HIPCHAT_ROOM                    - ID or name of the room to send the notification
+#   - HIPCHAT_URL                     - URL of the hipchat API  (defaults to v1 of the api)
+#
+# Example command line to run this playbook:
+#    ansible-playbook -vvvv -i "localhost," -c local \
+#       -e @overrides.yml \
+#       run_migrations.yml
```
Comment (maxrothman) [inline playbooks/continuous_delivery/rollback_migrations.yml]: Nit: fix the docs

**Review by maxrothman — COMMENTED** [review]
It seems like this gets called once and rolls back everything.

EDIT: if it wasn't clear, I was directly responding to [this comment](https://github.com/edx/configuration/pull/3502#issuecomment-276483455). We should get this straight before merging.

File: playbooks/continuous_delivery/rollback_migrations.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# This playbook will check for migrations to rollback for Django applications within a larger
+# Django application. The provided input file will determine what migrations should be rolled back.
```
Comment (feanil) [inline playbooks/continuous_delivery/rollback_migrations.yml]: `Django project` is how these are generally referred.

File: playbooks/continuous_delivery/rollback_migrations.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+#
+# The playbook uses the Django management commands found in this Django app repo:
+# https://github.com/edx/edx-django-release-util
+# So the Django app above needs to be installed in the Django app.
```
Comment (feanil) [inline playbooks/continuous_delivery/rollback_migrations.yml]: Again I think you mean `Django project' the second time you say `Django app`

**Comment by feanil** [comment]
A couple of small doc things but otherwise looks good.

**Review by doctoryes — COMMENTED** [review]
This file is a near copy of run_migrations.yml. Should the common parts be factored out into a common migrations_operation.yml file or the like?

**Comment by macdiesel** [comment]
@doctoryes I would prefer to keep them separate. Logically they perform two different tasks and there are some differences what files and data is input to complete each task. I think all we would end up with is a file with all the tasks combined and tags to differentiate them.

**Comment by macdiesel** [comment]
@maxrothman this playbook gets called once and rolls all the migrations in that file back.

**Review by doctoryes — APPROVED** [review]

**Pull request merged**